### PR TITLE
increase sshd security

### PR DESCRIPTION
**Pull request by fredsmith**
Suggestions from https://stribika.github.io/2015/01/04/secure-secure-shell.html

### CODE REVIEW

**Comment by jibsheet** [comment]
Can you confirm that this results in a machine reachable from os-x?  I know there were some issues around keys and ciphers in Sierra (mostly around retiring DSA, which we're also doing) but older os-x releases may have an old ssh client.  I'm not sure what our fleet runs, Wes might know.

I assume we can test by spinning up a sandbox with this branch pretty easily.

**Comment by fredsmith** [comment]
My concern about retiring the keys is that it causes the ssh host key to change, so people who have strict host key checking on will get a possibly unexpected warning after running this.  The rest of the changes are invisible to the client.

I can spin up a sandbox for testing.

**Comment by jibsheet** [comment]
Luckily, for most users, that's tools-edx-gp.edx.org and we can communicate it.

**Comment by fredsmith** [comment]
@jibsheet if you have the ability to test from the systems you have concerns about, fredsmith.sandbox.edx.org is up with this branch.   https://sshcheck.com/server/fredsmith.sandbox.edx.org/

File: playbooks/roles/vhost/tasks/main.yml [hunk 1]
```diff
@@ -76,21 +76,82 @@
...
     state: present
     regexp: "^PasswordAuthentication .*"
     line: "PasswordAuthentication {{ COMMON_SSH_PASSWORD_AUTH }}"
   register: sshd_config_line2
 
+- name: secure ssh algorithms
+  lineinfile:
+    name: /etc/ssh/sshd_config
+    regexp: '^KexAlgorithms'
+    insertafter: '^ServerKeyBits'
```
Comment (smarnach) [inline playbooks/roles/vhost/tasks/main.yml:91]: `ServerKeyBits` is for SSH protocol version 1, and not supported anymore in the OpenSSH version included in Ubuntu 17.04, so I suggest inserting the additional lines in a different place (or simply at the end of the file).

**Review by smarnach — COMMENTED** [review]
@fredsmith Thanks for pinging me.  I had a quick look at the PR, but I haven't yet verified whether this will cause any problems for us.  I will try to do so by Monday.

**Comment by jibsheet** [comment]
Should we be concerned about fixing 
```
ssh-rsa	RSA with SHA-1 hash 
```
while we're here?

I'll poll developers for old os-x (I have sierra and it's fine with that).

**Comment by smarnach** [comment]
I tested this, and this is unlikely to cause any problems for us, so all good from our side.

I'm not a cryptography expert, so it would take me a significant amount of time to judge the quality of the recommendations on https://stribika.github.io/2015/01/04/secure-secure-shell.html.  In cases like these, when I'm not really competent to judge something myself, I try to judge the party that is trying to sell it to me.  However, I couldn't find out anything about the author of the guide you linked, not even their real name.

How about following the recommendations on https://wiki.mozilla.org/Security/Guidelines/OpenSSH instead?  The Mozilla recommendations come from a reputed security team and seem more up to date (though they might be too restrictive, not sure).

File: playbooks/roles/vhost/tasks/main.yml (hunk 1 above)
Comment (myvcc657) [reply playbooks/roles/vhost/tasks/main.yml:91]: Perbaiki

**Comment by myvcc657** [comment]
`reslove all erroe file`

**Pull request closed without merging**